staging/src/k8s.io/pod-security-admission/admission/admission_test.go

```diff
@@ -278,7 +278,7 @@ func TestValidateNamespace(t *testing.T) {
 			expectAllowed:    true,
 			expectListPods:   false,
 			expectWarnings: []string{
-				`namespace "test" is exempt from Pod Security, and the policy (enforce=restricted:latest) will be ignored`,
+				`namespace "test" is exempt from Pod Security, and the policy (enforce=restricted:latest, warn=restricted:latest) will be ignored`,
```

Comment:
it's a bit irritating that this is showing up here... I'm wondering if the "default warning level to explicit enforce level" behavior should be opted into when calling PolicyToEvaluate:
* When admitting namespaces, I don't think we'd opt in (we only care about enforce levels there, and we probably don't want to include implicit defaulting when printing the policy we're ignoring)
* When admitting pod controllers, I think we'd opt in
* When admitting pods, I'm on the fence

Reply:
Yeah, I agree. I instead propose we update this warning to only print modes that have been explicitly set (if either the level or version is set, print both). I added this as a second commit, LMK what you think.

Regarding your suggestion to opt-in the defaulting behavior:
- when admitting namespaces, warn is only used when printing this error, which the explicit labeling check resolves
- when admitting pods, this should be a no-op most of the time. The only time it would do anything is if the `warn-version` is set (to something different from `enforce-version`), but not `warn` level (and `enforce` is set). In this case, the warn level would be evaluated at the different version iff it passed enforce. this seems like the expected behavior to me.

Reply:
yeah, that sounds good